types/smokehouse.d.ts

```diff
@@ -41,6 +42,8 @@
     retries?: number;
     /** A function that runs Lighthouse with the given options. Defaults to running Lighthouse via the CLI. */
     lighthouseRunner?: LighthouseRunner;
+    /** A function that gets a list of URLs requested to the server since the last fetch. */
+    takeNetworkRequestUrls?: () => string[];
```

Comment:
Does this need to be required? It's different than the other options, but if it's not provided then the `perf` smoke tests will always fail.

Another option would be to prune `networkRequests` if no `takeNetworkRequestUrls` is provided, but that is slightly more prone to accidentally not passing it in and the tests silently accepting that and passing anyways

A big question would be if @connorjclark feels like getting the network requests is doable with smokerider. If yes then required seems good. If not, we could go either way (e.g. it could still be required but smokerider could pass in a no-op function and [prune out the networkRequests itself in the `lib.js` frontend](https://github.com/GoogleChrome/lighthouse/blob/ba53dafc20fcc2f89322523c3666d80066538f20/lighthouse-cli/test/smokehouse/frontends/lib.js#L27-L48)).

Reply:
currently for smokerider the static server is run in a separate process, so `takeNetworkRequestUrls` wouldn't work. looking at the code again I see no reason for it being a different process, so should be possible to make this work.

Reply:
I've made it required for now.